Common/src/Storage/IResultTable.cs

```diff
@@ -126,6 +128,19 @@ IAsyncEnumerable<T> GetResults<T>(Expression<Func<Result, bool>> filter,
                                     Expression<Func<Result, T>>    convertor,
                                     CancellationToken              cancellationToken = default);
 
+  /// <summary>
+  ///   Get the results from a filter and convert it in the given type
+  /// </summary>
+  /// <param name="filter">MongoDB filter definition to select results</param>
+  /// <param name="convertor">Expression to convert result into another type</param>
+  /// <param name="cancellationToken">Token used to cancel the execution of the method</param>
+  /// <returns>
+  ///   Collection of results metadata from the database
+  /// </returns>
+  IAsyncEnumerable<T> GetResults<T>(FilterDefinition<Result>    filter,
+                                    Expression<Func<Result, T>> convertor,
+                                    CancellationToken           cancellationToken = default);
```

Comment:
This is not related to the PR.
Also, you should not expose MongoDB types in the interface.